### PR TITLE
Add foods to meals from diary

### PR DESCRIPTION
**Pull request by TheKaterTot**
They ask for ugly JavaScript, I deliver. This adds the list of foods to the diary page, a link to create a new food, and the ability to add a food to a meal. Calories update when the food is added to the meal.

### CODE REVIEW

**Comment by TheKaterTot** [comment]
I'm hoping to have time to refactor this tomorrow.

File: index.html [hunk 1]
```diff
@@ -143,6 +143,30 @@ <h4 class="center">Totals</h4 class="center">
             </tbody>
           </table>
         </div>
+
+        <div class="column column-50">
+          <h4 class="center">Foods</h4 class="center">
+            <a class="button button-primary" href="https://cmacaulay.github.io/quantified-self/foods.html">Add New Food</a>
```
Comment (cmacaulay) [inline index.html:149]: 🎉

File: index.html [hunk 2]
```diff
@@ -143,6 +143,30 @@ <h4 class="center">Totals</h4 class="center">
...
+              </tr>
+            </thead>
+            <tbody class="foods">
+            </tbody>
+          </table>
+          <h5>Add Selected to Meal</h5>
+          <input type="submit" class="button button-primary add-food-to-meal" data-category="breakfast" value="Breakfast">
+          <input type="submit" class="button button-primary add-food-to-meal" data-category="lunch" value="Lunch">
+          <input type="submit" class="button button-primary add-food-to-meal" data-category="snacks" value="Snacks">
+          <input type="submit" class="button button-primary add-food-to-meal" data-category="dinner" value="Dinner">
```
Comment (cmacaulay) [inline index.html:167]: 🎉 buttons 🔘

File: lib/index.js [hunk 3]
```diff
@@ -144,3 +146,42 @@ function calorieColorWarnings(id, difference) {
...
+              }
+
+  let options = { breakfast: fetchBreakfast,
+                  lunch: fetchLunch,
+                  dinner: fetchDinner,
+                  snacks: fetchSnacks }
+
+  $.post(`${host}/meals`, meal, () => {
+    options[category](currentDate);
+    $('input:checked').prop('checked', false);
```
Comment (cmacaulay) [inline lib/index.js:184]: this looks so complicated! I would love to go through this with you tomorrow.

File: main.bundle.js [hunk 4]
```diff
@@ -61,6 +61,7 @@
 	  fetchDiaryInfo();
```
Comment (cmacaulay) [inline main.bundle.js:61]: 🎉 `npm run build`

**Review by cmacaulay — APPROVED** [review]
Looks great! Thank you for all of your hard work on this, I know it was a 🤕

**Pull request merged**